independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/ClassLoadingResource.java

```diff
@@ -5,13 +5,21 @@
 
 public interface ClassLoadingResource {
 
+    interface BufferSpan {
```

Comment:
Could you add some javadoc / comments here? 

w/o reading the PR conversation it's going to be hard to figure out what's going on..